lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py

```diff
@@ -327,26 +329,43 @@ def build_entity(self):
             ) if storage_type is not None else None
         )
 
-    def _attached_sds_service(self):
+    def _attached_sds_service(self, dc_name):
         # Get data center object of the storage domain:
         dcs_service = self._connection.system_service().data_centers_service()
 
-        # Serach the data_center name, if it does not exists, try to search by guid.
-        dc = search_by_name(dcs_service, self._module.params['data_center'])
+        # Search the data_center name, if it does not exists, try to search by guid.
+        dc = search_by_name(dcs_service, dc_name)
         if dc is None:
-            dc = dcs_service.service(self._module.params['data_center']).get()
+            dc = dcs_service.service(dc_name).get()
             if dc is None:
                 return
 
         dc_service = dcs_service.data_center_service(dc.id)
         return dc_service.storage_domains_service()
 
-    def _maintenance(self, storage_domain):
-        attached_sds_service = self._attached_sds_service()
-        if attached_sds_service is None:
-            return
+    def _attached_sd_service(self, storage_domain):
+        dc_name = self._module.params['data_center']
+        # If datacenter parameter is missing (case: state == absent) iterate over all dcs
+        if not dc_name and self._module.params['state'] == 'absent':
+            dcs_service = self._connection.system_service().data_centers_service()
```

Comment:
I would re-write to something like this:

```python
def _attached_sd_service(self, storage_domain):
  dc_name = self._module.params['data_center']
  # If datacenter parameter is missing (case: state == absent) iterate over all dcs
  if not dc_name and self._module.params['state'] == 'absent':
    dcs_service = self._connection.system_service().data_centers_service()
    dc = search_by_attributes(dcs_service, storage=self.param('name'), status='up')
    if dc is None:
        raise Exception("Can't remove storage, because dataceneter bla bla')
  attached_sds_service = self._attached_sds_service(dc_name)
  attached_sd_service = attached_sds_service.storage_domain_service(storage_domain.id)
  return attached_sd_service
```